src/platform/macos/mod.rs

```diff
@@ -627,6 +631,55 @@ impl Window {
     }
 }
 
+unsafe fn nswindow_set_min_dimensions<V: NSWindow + Copy>(
+    window: V, min_width: f64, min_height: f64)
+{
+    window.setMinSize_(NSSize {
+        width: min_width,
+        height: min_height,
+    });
+    // If necessary, resize the window to match constraint
```

Comment:
Unlike the previous strategy, calling `setMinSize` does *not* change the window size if the window is smaller than the minimum size. In this case, if the user attempts to resize, the window will immediately expand to the minimum size, which is not a great UX, so I decided to update the window size if necessary.